kubernetes/hypertrace.sh

```diff
@@ -40,15 +40,51 @@ error_report() {
 trap 'error_report ${LINENO} ${BASH_COMMAND}' ERR
 
 function usage() {
-    echo "usage: $script {install|uninstall} [option]"
+    echo "usage: $script {install|uninstall|generate-manifests} [option]"
     echo " "
-    echo "available options:"
+    echo "available options for install:"
     echo " "
     echo "--clean             removes previous deployments of Hypertrace and do clean install"
+    echo "available options for generate-manifests:"
+    echo " "
+    echo "servicename          enter name of the service for which you want to generate manifests"
+    echo " "
+    echo "pre-install-tasks    generates manifests for services with pre-install helm hook"
+    echo " "
+    echo "post-install-tasks    generates manifests for services with post-install helm hook"
 
     exit 1
 }
 
+function cleanup(){
+    echo "[INFO] cleaning up any helm temporary working directory"
+    rm -rf ${HYPERTRACE_HOME}/data-services/charts
+    rm -rf ${HYPERTRACE_HOME}/data-services/tmpcharts
+    rm -rf ${HYPERTRACE_HOME}/data-services/Chart.lock
+    rm -rf ${HYPERTRACE_HOME}/platform-services/charts
+    rm -rf ${HYPERTRACE_HOME}/platform-services/tmpcharts
+    rm -rf ${HYPERTRACE_HOME}/platform-services/Chart.lock
+}
+
+function cleanup_manifests(){
+  echo "[INFO] removing old manifest files"
+  rm -rf ${HYPERTRACE_HOME}/data-services/helm-deployment-templates
+  rm -rf ${HYPERTRACE_HOME}/platform-services/helm-deployment-templates
+}
+
+function create_manifest_directories(){
+    mkdir ${HYPERTRACE_HOME}/data-services/helm-deployment-templates/
+    mkdir ${HYPERTRACE_HOME}/platform-services/helm-deployment-templates/
+}
+
+function create_helm_install_manifests(){
+  if [ $HT_DATA_STORE == "postgres" ]; then
+    helm install --dry-run hypertrace-platform-services ${HYPERTRACE_HOME}/platform-services/ -f ${HYPERTRACE_HOME}/platform-services/values.yaml -f ${HYPERTRACE_HOME}/platform-services/postgres/values.yaml -f ${HYPERTRACE_HOME}/clusters/$HT_PROFILE/values.yaml --set htEnv=${HT_ENV} > ${HYPERTRACE_HOME}/platform-services/helm-deployment-templates/install-manifests.yaml
```

Comment:
This should be `helm template`

Reply:
This works fine with `docker-desktop`, but I see it creates some dependency.

Reply:
Actually, this works even without k8s as long as you are using helm v3 as helm 3 removed the dependency on connection to tiller which works inside k8s.

I have actually tried this on my machine with k8s turned off and this works. And as we use chart `apiVersion: v2` and helm 3 across our setup and repositories we should be good.